xonsh/tools.py

```diff
@@ -1686,15 +1686,24 @@ def globpath(s, ignore_case=False, return_empty=False):
 
 def _iglobpath(s, ignore_case=False):
     s = builtins.__xonsh_expand_path__(s)
+    glob_sorted = builtins.__xonsh_env__.get('GLOB_SORTED')
     if ignore_case:
         s = expand_case_matching(s)
     if sys.version_info > (3, 5):
         if '**' in s and '**/*' not in s:
             s = s.replace('**', '**/*')
         # `recursive` is only a 3.5+ kwarg.
-        return glob.iglob(s, recursive=True), s
+        if glob_sorted:
+            paths = sorted(glob.glob(s, recursive=True))
```

Comment:
This should actually return an iterator, since it is an iglob replacement.  Something that is a bit more efficient would be:

```python
paths = glob.glob(s, recursive=True)
paths.sort()
paths = iter(paths)
```

I think that will work.